src/foam/nanos/notification/NotificationView.js

```diff
@@ -8,24 +8,36 @@ foam.CLASS({
   package: 'foam.nanos.notification',
   name: 'NotificationView',
   extends: 'foam.u2.View',
-  axioms: [ foam.pattern.Faceted.create() ],
 
-  properties: [ 'of' ],
+  axioms: [
+    foam.pattern.Faceted.create()
+  ],
+
+  properties: [
+    'of'
+  ],
+
   methods: [
     function initE() {
       this.SUPER();
       this
         .addClass(this.myClass())
-        .start('div').on('click', this.onClick).addClass('msg').add(this.data.body).end();
+        .start('div')
+          .on('click', this.onClick)
+          .addClass('msg')
+          .add(this.data.body)
+          .show(this.data.body !== '')
+        .end();
     }
   ],
+
   listeners: [
-    function onClick () {
-      if ( this.childNodes[0].css.display == "block" ) {
-        this.childNodes[0].style({display:'-webkit-box'})
+    function onClick() {
```

Comment:
Referencing childNodes[0] is fragile. Better to toggle a boolean property and then bind the style to the slot with a map at the creation side so that this code doesn't need to be changed if the child changes its location.